app/prism/PrismLogic.scala

```diff
@@ -116,12 +116,12 @@ object PrismLogic {
     for {
       (t, instances) <- amisWithInstances.toMap
       ssa <- ssas
-      instancesCount = instances.filter(i => doesInstanceBelongToSSA(i, ssa)).length
-      if(instancesCount > 0)
+      instancesCount = instances.count(i => doesInstanceBelongToSSA(i, ssa))
+      if instancesCount > 0
```

Comment:
I am not sure I like this syntax to be honest. I don't feel strongly though so up to you